### PR TITLE
Feature/ens - ready for review

### PR DESCRIPTION
**Pull request by Seroxdesign**

### CODE REVIEW

**Comment by Seroxdesign** [comment]
Closes:

https://github.com/MetaFam/TheGame/pull/1513

**Comment by dysbulic** [comment]
https://user-images.githubusercontent.com/181523/219142278-3738f0d3-88ed-41be-81ac-f8b26b4c811b.mp4

**Comment by Seroxdesign** [comment]
> 

I added a check for username, this issue should be resolved now.

**Comment by alalonde** [comment]
@Seroxdesign Could you please squash some of these commits? At least the testing ones (presumably that's what all the up/down.sql changes were)

**Comment by alalonde** [comment]
I did a bit of testing, including finding players with ENS names and usernames and vice versa. Everything looks to be working from my minimal testing

**Comment by alalonde** [comment]
> Could you please squash some of these commits?

Actually, disregard this, I didn't realize commits were squashed automatically on merge

**Comment by Seroxdesign** [comment]
> > Could you please squash some of these commits?
> 
> Actually, disregard this, I didn't realize commits were squashed automatically on merge

I'm assuming this is no longer an issue, so I won't be squashing any commits.

**Review by alalonde — CHANGES_REQUESTED** [review]

File: packages/design-system/vite.config.ts.timestamp-1675691765700.mjs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// vite.config.ts
+import react from "file:///home/sero/TheGame/node_modules/@vitejs/plugin-react/dist/index.mjs";
```
[the remark this answers is not in the document]
Comment (Seroxdesign) [reply packages/design-system/vite.config.ts.timestamp-1675691765700.mjs:2]: @alalonde this popped up while testing, will delete, not relevant to PR

File: hasura/metadata/tables.yaml [hunk 2]
```diff
@@ -481,6 +481,9 @@
       total_xp: totalXP
       profile_layout: profileLayout
   object_relationships:
+  - name: PlayerRank
+    using:
+      foreign_key_constraint_on: rank
```
[the remark this answers is not in the document]
Comment (Seroxdesign) [reply hasura/metadata/tables.yaml]: Changes were reversed. all happened during testing @alalonde

**Review by alalonde — APPROVED** [review]

**Pull request closed without merging**